src/units/abilities.cpp

```diff
@@ -1188,6 +1194,15 @@ static std::optional<std::string> select_damage_type(const unit_ability_list& ab
 				return type_count[lhs] < type_count[rhs];
 			});
 		}
+		if(is_alternative){
+			if(other_){
+				std::sort( std::begin( type_list ) , std::end( type_list ) , [&]( const std::string& rhs , const std::string& lhs ){
+					return (*other_).resistance_ability(resistance_abilities, lhs) < (*other_).resistance_ability(resistance_abilities, rhs);
+				});
+			} else {
+				type_list.clear();
```

Comment:
What is this for?

Reply:
in report.cpp opponent don't exist and can't be checked,in that case instead ot show an type who don't matche with an damage_type used against a opponent x, i prefer don't show alternative_typewhen the list contain many different

Reply:
Seems like the code path from the report should not end up here. The report should simply list all alternative types IMO.

Reply:
I can't add type to infinite in report

Reply:
If you mean the display of the report can overflow the available space then that is an issue of the report. Pretty sure it will already be handled appropriately and show an ellipsis in that case. For the icons we could add a limit of 3 or something.

I wouldn't worry too much about extreme cases anyway. The norm is still going to be overwhelmingly a single attack type and occasionally one alternate type. If a case of 5 alternate types looks ugly that is not that big a deal.

Reply:
icons added when multiple alternative_type

Reply:
I don't mind an icon for when there are too many types but this is not the place to implement that. That logic belongs into the report code. Please just remove the alternative_type handling from `select_replacement_type_or_alternative_without_resistance` and rename it.